### PR TITLE
[Fixes #273] Workflow for pytest

### LINKED ISSUE #273: CI: GitHub action for BackEndApp
## Feature Request

Is your feature request related to an existing [Issue](https://www.github.com/Auto-DL/Generator/issues) or a new implementation?

All test related issues

### Feature Description

Add a CI using GitHub actions for testing the Django project and it's unit tests

### PR DESCRIPTION
**Pull request by bamblebam**
# Pull Request

## What does this PR do?
Fixes #273 

Add github workflow to check python tests via pytest.

## What part does this affect?
- [ ] FrontEnd.
- [ ] BackEnd.
- [ ] Documentation.
- [x] Other. (Please specify below)
CI/CD

## Before submitting
- [x] Was this discussed/approved via a GitHub issue or slack?
- [x] Did you read the [contributor guideline](https://github.com/Auto-DL/Auto-DL/blob/v1-beta/CONTRIBUTING.md)?
- [x] Did you ensure that there aren't any other open [Pull Requests](https://www.github.com/Auto-DL/Generator/pulls) for the same update/change?
- [x] Did you make sure the title is self-explanatory and the description concisely explains the PR?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure the code is clean and docstrings have been added or updated as required?
- [ ] Did you make sure the code is linted/formatted locally prior to submission? (using `black` and/or `prettier`)
- [ ] Did you make sure to update the documentation with your changes? (if necessary)

## PR review
Anyone in the community is free to review the PR once the tests have passed.

Thank you for contributing to [AutoDL](https://github.com/Auto-DL/Auto-DL). We look forward to your continued support.

### CODE REVIEW

**Comment by RusherRG** [comment]
@bamblebam there is one check failing just try to fix it in the PR. Also, try to figure out why it hasn't been detected so far. Probably, might be because of a new update to CodeQL

**Review by RusherRG — CHANGES_REQUESTED** [review]
@bamblebam there is one check failing just try to fix it in the PR. Also, try to figure out why it hasn't been detected so far. Probably, might be because of a new update to CodeQL

**Comment by bamblebam** [comment]
@RusherRG I fixed it by commenting the line that was causing the issue. If it doesn't break anything we can remove it later.

**Review by RusherRG — APPROVED** [review]

**Review by RusherRG — CHANGES_REQUESTED** [review]
Just have a look at frontend-CI workflow and accordingly change the on event trigger and the name of the workflow to Backend CI

**Review by RusherRG — APPROVED** [review]

**Review by ADI10HERO — CHANGES_REQUESTED** [review]
Can we add caching please?

**Comment by bamblebam** [comment]
@ADI10HERO what do we want to cache ?

**Comment by ADI10HERO** [comment]
@bamblebam 
Hey, good question...

Python dependencies so the subsequent builds are faster till the cache is valid.
[Reference](https://github.com/actions/cache/blob/main/examples.md#python---pip)

**Comment by bamblebam** [comment]
alright

File: .github/workflows/backend-ci.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: "Backend CI"
+on: [push, pull_request]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.8]
```
Comment (ADI10HERO) [inline .github/workflows/backend-ci.yml:8]:
```suggestion
        python-version: [3.8]
    defaults:
      run:
        working-directory: ./BackEndApp
```

File: .github/workflows/backend-ci.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+          python-version: ${{ matrix.python-version }}
+      - uses: actions/cache@v2
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
+      - name: Install dependencies
+        run: |
+          cd BackEndApp
```
Comment (ADI10HERO) [inline .github/workflows/backend-ci.yml:23]:
```suggestion
```

**Review by ADI10HERO — CHANGES_REQUESTED** [review]
I hope the changes are self-explanatory!
If not please feel free to start a convo on slack :)

**Review by ADI10HERO — APPROVED** [review]

**Pull request merged**